tests/integration/test_storage_kafka/test.py

```diff
@@ -407,7 +435,21 @@ def test_kafka_formats(kafka_cluster):
         #         # ''
         #     ],
         # },
-        # TODO: test for AvroConfluence
+        'AvroConfluent' : {
+            'data_sample': [
+                avro_confluent_message(cluster.schema_registry_client, {'id':0L,'blockNo':0,'val1':unicode('AM'),'val2':0.5,"val3":1}),
+
+                ''.join(map(lambda id: avro_confluent_message(cluster.schema_registry_client, {'id':id,'blockNo':0,'val1':unicode('AM'),'val2':0.5,"val3":1}), range(1,16))),
+
+                avro_confluent_message(cluster.schema_registry_client, {'id':0L,'blockNo':0,'val1':unicode('AM'),'val2':0.5,"val3":1}),
+
+                '',
```

Comment:
currently 'empty value' will not fail the test because it is the last message on kafka topic (the error would happen after we have received all expected rows).


we should probably send 'empty value' first  + shift offsets in `expected` TSV by +1.

Reply:
I don't understand why this test should not fail, but I've verified this w/o this PR and indeed, it does not fail, however it should (otherwise what is the purpose of the test?)

Reply:
Updated the code to test empty messages for formats that support it